### PR TITLE
Lynx: migrate root from goerli to sepolia

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [X] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**

Goerli will be deprecated by end of year.

Prepare shift of root contracts on Lynx to use Sepolia instead of Goerli. The child contract deployments remain the same, so should have no impact on existing rituals etc. It should be a relatively simple transition of updating the lynx contract registry in `nucypher` and updating lynx nodes to use sepolia endpoints.

- Updated root constructor parameters based on latest used for mainnet deployment
- Updated root parameters / scripts to use Sepolia instead of Goerli
- Deployed Lynx contracts to Sepolia and update root part of contract registry (remove entries for chain 5)
- Verified contracts on Sepolia
    - [LynxStakingToken](https://sepolia.etherscan.io/address/0x347370278531Db455Aec3BFD0F30d57e41422353#code)
    - [MockPolygonRoot](https://sepolia.etherscan.io/address/0xDD60a8E632c13fb777Ca76C7FE5670031202Edab#code)
    - [TACo Application (Proxied)](https://sepolia.etherscan.io/address/0x329bc9Df0e45f360583374726ccaFF003264a136#code)
    - [TestnetThresholdStaking](https://sepolia.etherscan.io/address/0x18006f9A84C0bAD4CD96Aa69C7cE17aD760cDaD2#code)
- Executed `configure_staking` script on Sepolia of Lynx nodes
- Executed `confirm_operator_address` script on Sepolia for Lynx nodes.

# Follow-up tasks
- `nucypher` PR to update lynx contract registry and deprecate use of goerli - <https://github.com/nucypher/nucypher/pull/3376>
- Update lynx nodes eth-endpoint value to use sepolia instead of goerli.
- `nucypher-porter` PR that relocks porter's dependencies to use updated `nucypher` dependency with updated contract registry - TBD
- Update porter lynx eth-endpoint value to use sepolia instead of goerli.

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

**Review by vzotova — APPROVED** [review]

File: scripts/lynx/upgrade_root.py [hunk 1]
```diff
@@ -10,16 +11,13 @@
 
 def main():
     """
-    This script upgrades TACoApplication in Lynx Goerli.
+    This script upgrades TACoApplication in Lynx Sepolia.
     """
 
     deployer = Deployer.from_yaml(filepath=CONSTRUCTOR_PARAMS_FILEPATH, verify=VERIFY)
 
     taco_app_proxy_address = "0x31F0a0B94C829Ba6d902c98CAF8d8462C6c63241"
```
Comment (KPrasch) [inline scripts/lynx/upgrade_root.py:19]: Does this address need updating?

File: deployment/constructor_params/lynx/upgrades-root.yml [hunk 2]
```diff
@@ -10,20 +10,22 @@ constants:
...
   TEN_MILLION_TOKENS_IN_WEI_UNITS: 10000000000000000000000000
   YEAR_2025: 1735689600
 
 contracts:
   - TACoApplication:
       constructor:
-        _token: "0x90990F1F7C952D4D02759802565d62479e8aA936"
-        _tStaking: "0xcdD63981c8f4a2A684d102f7d7693A1c326CDd33"
+        _token: "0x347370278531Db455Aec3BFD0F30d57e41422353"
+        _tStaking: "0x18006f9A84C0bAD4CD96Aa69C7cE17aD760cDaD2"
```
Comment (KPrasch) [inline deployment/constructor_params/lynx/upgrades-root.yml:25]: Verified 👍🏻 

<https://sepolia.etherscan.io/address/0x347370278531Db455Aec3BFD0F30d57e41422353#code>

<https://sepolia.etherscan.io/address/0x18006f9A84C0bAD4CD96Aa69C7cE17aD760cDaD2#code>

File: scripts/lynx/upgrade_root.py (hunk 1 above)
Comment (KPrasch) [reply scripts/lynx/upgrade_root.py:19]:
```suggestion
    taco_app_proxy_address = "0x329bc9Df0e45f360583374726ccaFF003264a136"
```

**Review by KPrasch — APPROVED** [review]
Just the one suggestion, otherwise looks good 🌮

**Pull request merged**